iroh/src/endpoint.rs

```diff
@@ -875,49 +875,25 @@ impl Endpoint {
     }
 
     /// Returns the current [`NodeAddr`].
+    /// As long as the endpoint was able to binde to a network interfaces, some
+    /// local addresses will be available.
     ///
-    /// If this is called right after the [`Endpoint`] has been created, there is
-    /// a good chance these fields will be empty.
-    ///
+    /// The state of other fields depends on the state of networking and connectivity.
     /// Use the [`Endpoint::online`] method to ensure that the endpoint is considered
-    /// "online" (has contacted a  relay server and has at least one local addresses)
-    /// before calling this method, if you want to ensure that the `NodeAddr` will
-    /// contain enough information to allow this endpoint to be dialable by
-    /// a remote endpoint.
-    ///
-    /// Or, use the [`Endpoint::watch_node_addr`] method to get updates when the
-    /// `NodeAddr` changes.
+    /// "online" (has contacted a relay server) before calling this method, if you want
+    /// to ensure that the `NodeAddr` will contain enough information to allow this endpoint
+    /// to be dialable by a remote endpoint over the internet.
     ///
-    /// The `NodeAddr` will change as:
-    /// - network conditions change
-    /// - the endpoint connects to a relay server
-    /// - the endpoint changes its preferred relay server
-    /// - more addresses are discovered for this endpoint
+    /// You can use the [`Endpoint::watch_node_addr`] method to get updates when the `NodeAddr`
+    /// changes.
     pub fn node_addr(&self) -> NodeAddr {
-        let node_id = self.node_id();
-        let relay_url = self.msock.home_relay().get().first().cloned();
-        let direct_addresses = self
-            .msock
-            .direct_addresses()
-            .get()
-            .unwrap_or_default()
-            .into_iter()
-            .map(|addr| addr.addr)
-            .collect();
-
-        NodeAddr {
-            node_id,
-            relay_url,
-            direct_addresses,
-        }
+        self.watch_node_addr().get()
```

Comment:
This kind of gets us back to the "why have two methods?" question.  This one is now pretty trivial.